cvat/apps/iam/permissions.py

```diff
@@ -100,6 +106,22 @@ class OpenPolicyAgentPermission(metaclass=ABCMeta):
     scope: str
     obj: Optional[Any]
 
+    @classmethod
+    @abstractmethod
+    def _get_scopes(cls, request: ExtendedRequest, view: ViewSet, obj: Any) -> list:
+        """Method to override to define scopes based on the request"""
+
+    @classmethod
+    def get_scopes(cls, request: ExtendedRequest, view: ViewSet, obj: Any):
+        with suppress(KeyError):
+            scopes = cls._get_scopes(request, view, obj)
+
+            # prevent code bugs when _get_scope defines scopes "softly"
```

Comment:
What do you mean by "softly"?

Reply:
![image](https://github.com/user-attachments/assets/ebb78222-5fad-4ee0-ae39-7466002a3ae0)

Reply:
Ah, okay. In that case, `assert all(scopes)` seems more appropriate, since returning a `None` scope is a bug.